packages/core/types/src/dml/index.ts

```diff
@@ -180,9 +180,20 @@ export type InferIndexableProperties<T> = keyof (T extends IDmlEntity<
     } & InferForeignKeys<T>
   : never)
 
-export type EntityIndex<TSchema extends DMLSchema = DMLSchema> = {
+export type EntityIndex<
+  TSchema extends DMLSchema = DMLSchema,
+  TWhere = string
```

Comment:
Suggestion : shouldn't it be string or record? And the keys for record are the inferInxableProperties wdyt?

Reply:
only at the entry point, everywhere else it'll be a string. I've added an override at the entry point

Reply:
Ok got it 😉